ext/dcos-installer/dcos_installer/cli.py

```diff
@@ -1,108 +1,163 @@
-        pp.stage_name = action.__name__
-        pp.beautify('print_data')
-
-    finally:
-        loop.close()
-    exitcode = 0
-    for host_result in result:
-        for command_result in host_result:
-            for host, process_result in command_result.items():
-                if process_result['returncode'] != 0:
-                    exitcode += 1
-    print_header('ACTION {} COMPLETE'.format(action.__name__))
-    pp.print_summary()
-    return exitcode
-
-
-def tall_enough_to_ride():
-    choices_true = ['yes', 'y']
-    choices_false = ['no', 'n']
-    while True:
-        do_uninstall = input('This will uninstall DC/OS on your cluster. You may need to manually remove '
-                             '/var/lib/zookeeper in some cases after this completes, please see our documentation '
-                             'for details. Are you ABSOLUTELY sure you want to proceed? [ (y)es/(n)o ]: ')
-        if do_uninstall.lower() in choices_true:
-            return
-        elif do_uninstall.lower() in choices_false:
-            sys.exit(1)
-        else:
-            log.error('Choices are [y]es or [n]o. "{}" is not a choice'.format(do_uninstall))
-
-
-def get_new_analytics(config_path=CONFIG_PATH):
-    return InstallerAnalytics(backend.get_config(config_path))
-
-
-def dispatch_action(args):
-    installer_analytics = get_new_analytics()
-
-    def preflight(args):
-        print_header("EXECUTING PREFLIGHT")
-        return action_lib.run_preflight
-
-    def postflight(args):
-        print_header("EXECUTING POSTFLIGHT")
-        return action_lib.run_postflight
-
-    def deploy(args):
-        print_header("EXECUTING DC/OS INSTALLATION")
-        return action_lib.install_dcos
-
-    def uninstall(args):
-        print_header("EXECUTING UNINSTALL")
-        tall_enough_to_ride()
-        return action_lib.uninstall_dcos
-
-    def install_prereqs(args):
-        print_header("EXECUTING INSTALL PREREQUISITES")
-        return action_lib.install_prereqs
-
-    for action in ['deploy', 'preflight', 'postflight', 'uninstall', 'install_prereqs']:
-        if getattr(args, action):
-            errors = run_loop(locals()[action](args), args)
-            if not args.disable_analytics:
-                installer_analytics.send(
-                    action=action,
-                    install_method="cli",
-                    num_errors=errors,
-                )
-            exit_code = 1 if errors > 0 else 0
-            sys.exit(exit_code)
+def setup_logger(options):
+    level = 'INFO'
+    if options.verbose:
+        level = 'DEBUG'
+    coloredlogs.install(
+        level=level,
+        level_styles={
+            'warn': {
+                'color': 'yellow'
+            },
+            'error': {
+                'color': 'red',
+                'bold': True,
+            },
+        },
+        fmt='%(message)s',
+        isatty=True
+    )
+    log.debug("Logger set to DEBUG")
+
+
+def start_installer(args=None):
```

Comment:
don't give a default to args here. We don't use it, it's dead code that could potentially hide programmer mistakes, and in theory makes it so every function this code passes args() to should actually check for `None` or list() to be correct.